pkg/virt-controller/watch/migration/migration.go

```diff
@@ -2084,17 +2079,18 @@ func (c *Controller) deleteVMI(obj interface{}) {
 	}
 }
 
-func (c *Controller) outboundMigrationsOnNode(node string, runningMigrations []*virtv1.VirtualMachineInstanceMigration) (int, error) {
+func (c *Controller) outboundMigrationsOnNode(node string, runningMigrations []*virtv1.VirtualMachineInstanceMigration) int {
 	sum := 0
 	for _, migration := range runningMigrations {
 		key := controller.NamespacedKey(migration.Namespace, migration.Spec.VMIName)
-		if vmi, exists, _ := c.vmiStore.GetByKey(key); exists {
-			if vmi.(*virtv1.VirtualMachineInstance).Status.NodeName == node {
-				sum = sum + 1
+		if obj, exists, _ := c.vmiStore.GetByKey(key); exists {
+			vmi := obj.(*virtv1.VirtualMachineInstance)
+			if vmi.Status.NodeName == node || (vmi.Status.MigrationState != nil && vmi.Status.MigrationState.SourceNode == node) {
```

Comment:
the vmi.migrationState can be there from a previous migration. we need to check the migration uuid.